cmdstanpy/utils.py

```diff
@@ -925,56 +925,77 @@ def read_rdump_metric(path: str) -> List[int]:
 def do_command(
     cmd: List[str],
     cwd: Optional[str] = None,
-) -> Optional[str]:
+    *,
+    show_console: bool = False,
+    fd_out: Optional[TextIO] = None,
+) -> None:
     """
-    Spawn process, print stdout/stderr to console.
-    Throws RuntimeError on non-zero returncode.
+    Run command as subprocess
+    Raises ``RuntimeError`` on non-zero return code or execption ``OSError``.
+
+    :param cmd: command and args.
+    :param cwd: directory in which to run command, if unspecified,
+        run command in the current working directory.
+    :param show_console: when ``True``, streams stdout to sys.stdout.
+        Default is ``True``.
+    :param fd_out: when specified, sends stdout to this stream as well.
     """
-    get_logger().debug('cmd: %s', cmd)
+    get_logger().debug('cmd: %s', ' '.join(cmd))
     try:
         proc = subprocess.Popen(
             cmd,
             cwd=cwd,
+            bufsize=1,
             stdin=subprocess.DEVNULL,
             stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
+            stderr=subprocess.STDOUT,  # avoid buffer overflow
```

Comment:
So what happened with the buffer overflow? This was due to C++ error output?

Reply:
Or could we just check both pipes when doing proc.poll ?

Reply:
C++ compiler error novels seems most likely.  Sending stderr to stdout greatly simplifies the logic and as we end of parsing both to get assorted error messages, there's no reason not to combine them.